packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -32,39 +36,64 @@ const propTypes = {
    */
   isStartDateRange: PropTypes.bool,
   /**
-   * A moment object to represents the maximum date that can be selected.
+   * An ISO 8601 string representation of the maximum date that can be selected.
    */
-  maxDate: PropTypes.object,
+  maxDate: PropTypes.string,
   /**
-   * A moment object to represents the minimum date that can be selected.
+   * An ISO 8601 string representation of the minimum date that can be selected.
    */
-  minDate: PropTypes.object,
+  minDate: PropTypes.string,
   /**
    * A callback function to execute when a valid date is selected or entered.
    */
   onChange: PropTypes.func,
   /**
-   * The selected date to show in the date input.
-   */
-  selectedDate: PropTypes.object,
-  /**
-   * A moment object to use as the default start date for a date range.
+   * An ISO 8601 string representation of the selected start date. startDate represents the currently selected date after some change has been made where as defaultDate represents the initial default date that shows in the input before any change is made.
    */
-  startDate: PropTypes.object,
+  startDate: PropTypes.string,
 };
 
+const locale = 'en-US'; // TODO: Get the locale from i18n
+const dateFormat = 'MM/DD/YYYY'; // TODO: Get the locale from i18n
+
 const defaultProps = {
   isEndDateRange: false,
   isStartDateRange: false,
 };
 
 class DatePicker extends React.Component {
+
+  static safeMoment(date) {
+    if (date && dateFormat) {
+      const momentDate = moment.utc(date, dateFormat);
+      return momentDate.isValid() ? momentDate : date;
+    }
+
+    return date;
+  }
+
+  static createMomentsFromISO8601(dates) {
+    const momentDates = [];
+
+    if (dates) {
+      let index = 0;
+      for (index = 0; index < dates.length; index += 1) {
+        const momentDate = moment.utc(dates[index], dateFormat);
+        if (momentDate.isValid()) {
+          momentDates.push(momentDate);
+        }
+      }
+    }
+
+    return momentDates.length > 0 ? momentDates : dates;
```

Comment:
What if some dates are valid, and other dates are not? Should we perhaps return some kind of mapping indicating the successful dates, and another indicating the failure ones?